python/python/lance/lance/indices/__init__.pyi

```diff
@@ -47,3 +50,24 @@ def transform_vectors(
     pq_codebook: pa.Array,
     dst_uri: str,
 ): ...
+
+class IndexSegmentDescription:
+    uuid: str
+    dataset_version: int
+    fragment_ids: list[int]
```

Comment:
nit: should this be a set?

Reply:
I don't see any harm in it but I also don't see the reasoning?  Is it because we assume users will want to do set-like operations on this?